src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php

```diff
@@ -37,6 +40,54 @@ class DoctrineOrmMapper extends BaseDoctrineRdfMapper
     /**
      * {@inheritDoc}
      *
+     * The ORM implementation for preparing the object
+     */
+    public function prepareObject(TypeInterface $type, $parent = null)
+    {
+        $object = parent::prepareObject($type);
+
+        if (null == $parent) {
+            throw new RuntimeException('You need a parent to create new objects');
+        }
+
+        /** @var \Doctrine\ORM\Mapping\ClassMetadata $metaData */
+        $metaData = $this->om->getClassMetaData(get_class($object));
+
+        $parentMappingField = $this->findParentMapping($parent, $metaData);
+
+        if(false !== $parentMappingField) {
+            $metaData->setFieldValue($object, $parentMappingField, $parent);
+            return $object;
+        }
+
+        throw new RuntimeException('parent could not be found/set for '
+                . get_class($object));
+    }
+
+
+    /**
+     * find the parent object's property which holds the collection entries
+     *
+     * @param $parent
+     * @param ClassMetadata $metaData metadata from the collection entry's entity
+     * @return bool|string
+     */
+    protected function findParentMapping($parent, ClassMetadata $metaData = null)
+    {
+        $parentClass = ClassUtils::getRealClass(get_class($parent));
+
+        foreach ($metaData->associationMappings as $mapping) {
+            if ($mapping['targetEntity'] == $parentClass) {
+                return $mapping['fieldName'];
+            }
+        }
+
+        return false;
```

Comment:
i suggest that we throw an exception instead, something like "No mapping for parent class $parentClass found in metadata of $metaData->getName"